src/js/core/widget/core/Spin.js

```diff
@@ -0,0 +1,629 @@
+			placeholder.classList.add(classes.PLACEHOLDER);
+			element.appendChild(placeholder);
+
+			this._ui.placeholder = placeholder;
+			return element;
+		};
+
+		prototype._setValue = function (value, enableChangeEvent) {
+			var self = this,
+				animation;
+
+			value = window.parseInt(value, 10);
+			self._ui.placeholder.textContent = value;
+
+			if (isNaN(value)) {
+				ns.warn("Spin: value is not a number");
+			} else if (value !== self.options.value) {
+				if (value >= self.options.min && value <= self.options.max || self.options.loop === "enabled") {
+					self._prevValue = self.options.value;
+					if (self.options.loop === "enabled") {
+						if (value > self.options.max) {
+							value = self.options.min;
+						} else if (value < self.options.min) {
+							value = self.options.max;
+						}
+					}
+					self.options.value = value;
+					// set data-value on element
+					self.element.dataset.value = value;
+
+					// stop previous animation
+					animation = self.state.animation[0];
+					if (animation !== null && animation.to !== animation.current) {
+						self._animation.stop();
+					}
+					// update status of widget
+					self._show(enableChangeEvent);
+				}
+			}
+		};
+
+		prototype._getValue = function () {
+			return this.options.value;
+		};
+
+		prototype._setMax = function (element, max) {
+			var options = this.options;
+
+			options.max = (max !== undefined) ? parseInt(max, 10) : 0;
+			this.length = options.max - options.min + 1;
+		};
+
+		prototype._setMin = function (element, min) {
+			var options = this.options;
+
+			options.min = (min !== undefined) ? parseInt(min, 10) : 0;
+			this.length = options.max - options.min + 1;
+		};
+
+		prototype._setLabels = function (element, value) {
+			var self = this;
+
+			self.options.labels = value.split(",");
+			self._refresh();
+		};
+
+		prototype._setModuloValue = function (element, value) {
+			this.options.moduloValue = (value === "enabled") ? "enabled" : "disabled";
+		};
+
+		prototype._setShortPath = function (element, value) {
+			this.options.shortPath = (value === "enabled") ? "enabled" : "disabled";
+		};
+
+		prototype._setLoop = function (element, value) {
+			this.options.loop = (value === "enabled") ? "enabled" : "disabled";
+		};
+
+		prototype._setDuration = function (element, value) {
+			this.options.duration = window.parseInt(value, 10);
+		};
+
+		prototype._setEnabled = function (element, value) {
+			var self = this;
+
+			self.options.enabled = (value === "false") ? false : value;
+			if (self.options.enabled) {
+				element.classList.add(classes.ENABLING);
+				window.setTimeout(function () {
+					element.classList.remove(classes.ENABLING);
+				}, ENABLING_DURATION);
+				element.classList.add(classes.ENABLED);
+				utilsEvents.on(document, "drag dragend", self);
+
+				// disable tau rotaryScroller the widget has own support for rotary event
+				ns.util.rotaryScrolling && ns.util.rotaryScrolling.lock();
```

Comment:
I think the rotary event is for wearable.
It it no problem in core widget?

Reply:
Good suggestion!
It would be good to move to the wearable profile.

Reply:
@singa2000, Do I move the rotary event in the next PR?

Reply:
Sure~